### DISCUSSION TITLE
ipsec.conf and swanctl.conf co-existence to use new options in swanctl.conf

### DISCUSSION THREAD

**Discussion by vdabbara** [strongswan/strongswan#242] [General]
Hello,
We have been using ipsec.conf till now for the s2s vpn functionality. However, we have a new requirement where we need to preserve the dscp/tos value from the client packets and copy it to the tunnel headers. I found that swanctl.conf has one such option. 

connections.<conn>.children.<child>.copy_dscp [out]
Whether to copy the DSCP (Differentiated Services Field Codepoint) header field to/from the outer IP header in tunnel mode. The value out only copies the field from the inner to the outer header, the value in does the opposite and only copies the field from the outer to the inner header when decapsulating, the value yes copies the field in both directions, and the value no disables copying the field altogether. Setting this to yes or in could allow an attacker to adversely affect other traffic at the receiver, which is why the default is out. Controlling this behavior is not supported by all kernel interfaces.

My question is, can we continue to use ipsec.conf for the connections just like we have been using and then use the settings in swanctl.conf to influence the copy_dscp behavior for the connection defined in ipsec.conf? We don't want to completely migrate to swanctl and vici interface in short term. 

Please advice.

**Comment by strongX509** [comment]
Hi,
mixing `ipsec.conf` and `swanctl.conf` isn't really a good idea since the first configuration file is managed by the `stroke` interface and the latter by the `vici` interface but it might work if you are careful which dedicated `vici` commands you use.

The use of `ipsec.conf` has been deprecated for many years now and thus we haven't back-ported new `vici`configuration options to `stroke`. With the forthcoming strongSwan 6.0 post-quantum version which will be released sometime in 2021  we will get rid of `ipsec.conf`, the `stroke` interface and the legacy ìpsec` and `starter` commands altogether.

**Comment by vdabbara** [comment]
Thank you for the response. It appears migrating to vici configurations is the right direction to move forward. We will do that. 
I haven't seen any examples or test cases in relation to testing dscp configurations. 
eg. connections.<conn>.dscp 
      connections.<conn>.children.<child>.copy_dscp
I did a quick test by setting those options and i did not see dscp value being reflected in the ESP packets 

root@c106d0cc3e50:/# tcpdump -nnei t45-host icmp -v
tcpdump: listening on t45-host, link-type EN10MB (Ethernet), capture size 262144 bytes
09:53:04.988131 42:26:ef:e6:90:5f > 42:26:ef:e6:90:5f, ethertype IPv4 (0x0800), length 1242: (tos 0xb8, ttl 63, id 56307, offset 0, flags [DF], proto ICMP (1), length 1228)
    192.168.1.2 > 192.168.5.1: ICMP echo request, id 22, seq 389, length 1208
09:53:06.012111 42:26:ef:e6:90:5f > 42:26:ef:e6:90:5f, ethertype IPv4 (0x0800), length 1242: (tos 0xb8, ttl 63, id 56518, offset 0, flags [DF], proto ICMP (1), length 1228)
    192.168.1.2 > 192.168.5.1: ICMP echo request, id 22, seq 390, length 1208
^C
2 packets captured
2 packets received by filter
0 packets dropped by kernel
root@c106d0cc3e50:/# tcpdump -nnei wan1 esp -v 
tcpdump: listening on wan1, link-type EN10MB (Ethernet), capture size 262144 bytes
09:53:15.228170 82:ec:b3:03:c6:91 > ae:d2:8d:4d:33:bd, ethertype IPv4 (0x0800), length 1314: (tos 0x0, ttl 64, id 51987, offset 0, flags [DF], proto ESP (50), length 1300)
    10.113.191.11 > 10.113.191.12: ESP(spi=0xcf9e4969,seq=0x1cf), length 1280
09:53:16.252160 82:ec:b3:03:c6:91 > ae:d2:8d:4d:33:bd, ethertype IPv4 (0x0800), length 1314: (tos 0x0, ttl 64, id 52214, offset 0, flags [DF], proto ESP (50), length 1300)
    10.113.191.11 > 10.113.191.12: ESP(spi=0xcf9e4969,seq=0x1d0), length 1280


connections {
   t45-1 {
      local_addrs  = 10.113.191.11
      remote_addrs = 10.113.191.12
      dscp = 100
      children {
         t45-1-child {
            start_action = trap
            copy_dscp = out
         }
      }
   }
}

**Reply by Thermi** [reply]
Do you have logs from the installation of the CHILD_SA?
